src/wiki/plugins/redlinks/mdx/redlinks.py

```diff
@@ -65,19 +44,30 @@ def get_class(self, el):
             return
         if url.scheme == "mailto":
             return
-        if url.scheme or url.netloc or url.path.startswith("/"):
-            # Contains a hostname or is an absolute link => external
+        if url.scheme or url.netloc:
+            # Contains a hostname or url schema ⇒ External link
             return self.external_class
         # Ensure that path ends with a slash
         relpath = url.path.rstrip("/") + "/"
-        target = urljoin_internal(self.my_urlpath.path, relpath)
-        if target is None:
-            # Relative path goes outside wiki URL space => external
+        try:
+            target = resolve(urljoin(self.my_urlpath, relpath))
+        except Resolver404:
+            # Broken absolute link
+            return self.external_class
```

Comment:
No, the broken slug should be connected to (at least the option of, resolve doesn't know whether article creation needs permissions) the article creation view, which is a valid view. Or am I wrong? If I'm right, this set of lines should be covered by the redlinks test with server root != wiki root only. I'll try to check.

(That is, until someone managed to hack the wiki so far that it doesn't even resolve wiki article creation urls – and then what are they using `redlinks` for anyway, this might as well be a dead external link.)